remodel/object_handler.py

```diff
@@ -74,7 +74,7 @@ def __iter__(self):
 
     def __len__(self):
         self._fetch_results()
-        return len(self.result_cache)
+        return len(list(self.result_cache))
```

Comment:
When `self.result_cache` is set, it is explicitly set to a `list`. Why have you wrapped it again with a `list`?